jabkit/src/main/java/org/jabref/toolkit/commands/GenerateCitationKeys.java

```diff
@@ -58,21 +116,43 @@ public void run() {
             System.out.println(Localization.lang("Regenerating citation keys according to metadata."));
         }
 
-        CitationKeyGenerator keyGenerator = new CitationKeyGenerator(
-                databaseContext,
-                argumentProcessor.cliPreferences.getCitationKeyPatternPreferences());
+        CitationKeyPatternPreferences existingPreferences = parentCommand.getParent().cliPreferences.getCitationKeyPatternPreferences();
+
+        CitationKeyPatternPreferences preferencesToUse = existingPreferences;
+
+        if (transliterate != null || pattern != null || avoidOverwrite != null ||
+                warnBeforeOverwrite != null || generateBeforeSaving != null || suffix != null ||
+                keyPatternRegex != null || keyPatternReplacement != null ||
+                unwantedCharacters != null || keywordDelimiter != null) {
+
+            preferencesToUse = new CitationKeyPatternPreferences(
+                    transliterate != null ? transliterate : existingPreferences.shouldTransliterateFieldsForCitationKey(),
+                    avoidOverwrite != null ? avoidOverwrite : existingPreferences.shouldAvoidOverwriteCiteKey(),
+                    warnBeforeOverwrite != null ? warnBeforeOverwrite : existingPreferences.shouldWarnBeforeOverwriteCiteKey(),
+                    generateBeforeSaving != null ? generateBeforeSaving : existingPreferences.shouldGenerateCiteKeysBeforeSaving(),
+                    suffix != null ? suffix : existingPreferences.getKeySuffix(),
```

Comment:
Keep `key` -- variable should be `keySuffix`. See other lines. They are consistent (somehow)